splunk_connect_for_snmp_traps/manager/mib_server_client.py

```diff
@@ -46,39 +43,26 @@ def get_translation(var_binds, mib_server_url):
         var_binds_list.append(var_bind)
         org_var_bind = '{oid}="{value}"'.format(oid=str(name), value=str(val))
         trap_event_string += " " + org_var_bind
+
     payload["var_binds"] = var_binds_list
     payload = json.dumps(payload)
 
-    # trap_event_string = payload
-
-    # Send the POST request to mib server
     headers = {"Content-type": "application/json"}
     endpoint = "translation"
     TRANSLATION_URL = os.path.join(mib_server_url.strip("/"), endpoint)
     logger.debug(f"[-] TRANSLATION_URL: {TRANSLATION_URL}")
 
     try:
-        # resp = requests.request("POST", TRANSLATION_URL, headers=headers, data=payload)
-        # use Session with Retry
-        retry_strategy = Retry(
-            total=3,
-            backoff_factor=1,
-            status_forcelist=[429, 500, 502, 503, 504],
-            method_whitelist=["GET", "POST"],
-        )
-        adapter = HTTPAdapter(max_retries=retry_strategy)
-        session = requests.Session()
-        session.mount("https://", adapter)
-        session.mount("http://", adapter)
-        resp = session.post(TRANSLATION_URL, headers=headers, data=payload)
-        # If varBinds gets translated, overide it with the translated one
-        if resp.status_code == 200:
-            trap_event_string = resp.text
-        if resp.status_code != 200:
-            logger.error(f"[-] Mib Server API Error with code: {resp.status_code}")
+        session = ClientSession()
+        resp = await session.post(TRANSLATION_URL, headers=headers, data=payload)
+        if resp.status == 200:
+            trap_event_string = await resp.text()
```

Comment:
why do we need await on resp.text()? Does it make some additional call or is that just a value returned from resp?

Reply:
text is an async method. I spent a few hours debugging to find out that this is not a field. And as this is async method it need to be awaited.